src/test/util/setup_common.cpp

```diff
@@ -108,6 +108,9 @@ static void ExitFailure(std::string_view str_err)
 BasicTestingSetup::BasicTestingSetup(const ChainType chainType, TestOpts opts)
     : m_args{}
 {
+    if constexpr (!G_FUZZING) {
+        SeedRandomForTest(SeedRand::FIXED_SEED);
+    }
```

Comment:
In commit "fuzz: Clarify that only SeedRandomStateForTest(SeedRand::ZEROS) is allowed" (fae63bf13033adec80c7e6d73144a21ea3cfbc6d)

What's are reasons for not writing this as:

```c++
SeedRandomForTest(G_FUZZING ? SeedRand::ZEROS : SeedRand::FIXED_SEED);
```

It seems like this could eliminate boilerplate from fuzz tests, and also make the test setup behavior more intuitive: always setting a deterministic seed instead of sometimes setting one and sometimes not.

(Asking after this PR was already merged in context of followup #31521, which I'm finding a little confusing)

Reply:
re: https://github.com/bitcoin/bitcoin/pull/31486#discussion_r1891383934

> So the boilerplate can't be removed

To clarify, in the context of #31521 specifically, I think the change I'm suggesting would have prevented the nondeterminism there, and would avoid the need for a `SeedRandomStateForTest(SeedRand::ZEROS);` call at the top of the `utxo_total_supply` test.

But the `utxo_total_supply` test is just an unusual test, because most fuzz tests use test setup objects that get shared across fuzz inputs, while `utxo_total_supply` creates a new test setup object for each input.